visit:locate options broken

<!--
Before opening an issue, just take into account that this is a completely free of charge and open source project.
I'm always happy to help and provide support, but some understanding will be expected.
I do this in my own free time, so expect some delays when implementing new features and fixing bugs, and don't take it personal if an issue gets eventually closed.
You may also be asked to provide tests or ways to reproduce reported bugs.
Try to be polite, and understand it is impossible for an OSS project to cover all use cases.

With that said, please fill in the information requested next. More information might be requested next (like logs or system configs).
-->

#### How Shlink is set-up

* Shlink Version: `shlink --version` says `--latest`, I use this container: https://hub.docker.com/layers/shlink/shlinkio/shlink/latest/images/sha256-476514540f1587b598c146f27b5a8393144044652c1f53dedd4917c674f9247e?context=explore
* PHP Version: x.y.z
* How do you serve Shlink: Docker image
* Database engine used: PostgreSQL

#### Summary

<!-- Provide a summary describing the problem you are experiencing. -->

The `-a` and `-r` options for visit:locale are unrecognised despite being present in `--help`

```
/etc/shlink # shlink visit:locate -a -r


 [WARNING] You are about to process the location of all existing visits your short URLs received.

           Since shlink saves visitors IP addresses anonymized, you could end up losing precision on some of your
           visits.

           Also, if you have a large amount of visits, this can be a very time consuming process. Continue at your own
           risk.


 Do you want to proceed? (yes/no) [no]:
 > yes


 [ERROR] The "-a" option does not exist.


/etc/shlink # shlink visit:locate -r -a


 [WARNING] You are about to process the location of all existing visits your short URLs received.

           Since shlink saves visitors IP addresses anonymized, you could end up losing precision on some of your
           visits.

           Also, if you have a large amount of visits, this can be a very time consuming process. Continue at your own
           risk.


 Do you want to proceed? (yes/no) [no]:
 > yes


 [ERROR] The "-r" option does not exist.


/etc/shlink # shlink visit:locate --retry --all


 [WARNING] You are about to process the location of all existing visits your short URLs received.

           Since shlink saves visitors IP addresses anonymized, you could end up losing precision on some of your
           visits.

           Also, if you have a large amount of visits, this can be a very time consuming process. Continue at your own
           risk.


 Do you want to proceed? (yes/no) [no]:
 > yes


 [ERROR] The "--retry" option does not exist.



```

#### Current behavior

<!-- How is it actually behaving (and it shouldn't)? -->

#### Expected behavior

The `--all` and `--retry` options for `visit:locate` work

<!-- How did you expected to behave? -->

#### How to reproduce

Try and use `-a` and `-r` options

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -33,7 +33,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com), and this 
 * [#1280](https://github.com/shlinkio/shlink/issues/1280) Dropped support for PHP 8.0
 
 ### Fixed
-* *Nothing*
+* [#1471](https://github.com/shlinkio/shlink/issues/1471) Fixed error when running `visit:locate` command with any extra parameter (like `--retry`).
 
 
 ## [3.1.2] - 2022-06-04

--- a/module/CLI/src/Command/Util/AbstractLockedCommand.php
+++ b/module/CLI/src/Command/Util/AbstractLockedCommand.php
@@ -14,7 +14,7 @@ use function sprintf;
 
 abstract class AbstractLockedCommand extends Command
 {
-    public function __construct(private LockFactory $locker)
+    public function __construct(private readonly LockFactory $locker)
     {
         parent::__construct();
     }

--- a/module/CLI/src/Command/Visit/LocateVisitsCommand.php
+++ b/module/CLI/src/Command/Visit/LocateVisitsCommand.php
@@ -17,6 +17,7 @@ use Shlinkio\Shlink\IpGeolocation\Exception\WrongIpException;
 use Shlinkio\Shlink\IpGeolocation\Model\Location;
 use Shlinkio\Shlink\IpGeolocation\Resolver\IpLocationResolverInterface;
 use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\ArrayInput;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
@@ -80,12 +81,12 @@ class LocateVisitsCommand extends AbstractLockedCommand implements VisitGeolocat
             );
         }
 
-        if ($all && $retry && ! $this->warnAndVerifyContinue($input)) {
+        if ($all && $retry && ! $this->warnAndVerifyContinue()) {
             throw new RuntimeException('Execution aborted');
         }
     }
 
-    private function warnAndVerifyContinue(InputInterface $input): bool
+    private function warnAndVerifyContinue(): bool
     {
         $this->io->warning([
             'You are about to process the location of all existing visits your short URLs received.',
@@ -103,7 +104,7 @@ class LocateVisitsCommand extends AbstractLockedCommand implements VisitGeolocat
         $all = $retry && $input->getOption('all');
 
         try {
-            $this->checkDbUpdate($input);
+            $this->checkDbUpdate();
 
             if ($all) {
                 $this->visitLocator->locateAllVisits($this);
@@ -166,7 +167,7 @@ class LocateVisitsCommand extends AbstractLockedCommand implements VisitGeolocat
         $this->io->writeln($message);
     }
 
-    private function checkDbUpdate(InputInterface $input): void
+    private function checkDbUpdate(): void
     {
         $cliApp = $this->getApplication();
         if ($cliApp === null) {
@@ -174,7 +175,7 @@ class LocateVisitsCommand extends AbstractLockedCommand implements VisitGeolocat
         }
 
         $downloadDbCommand = $cliApp->find(DownloadGeoLiteDbCommand::NAME);
-        $exitCode = $downloadDbCommand->run($input, $this->io);
+        $exitCode = $downloadDbCommand->run(new ArrayInput([]), $this->io);
 
         if ($exitCode === ExitCodes::EXIT_FAILURE) {
             throw new RuntimeException('It is not possible to locate visits without a GeoLite2 db file.');

